tests/src/performance/perflab/LowLevelPerf.cs

```diff
@@ -0,0 +1,765 @@
+        MyDelegate aStaticDelegate = new MyDelegate(Class.EmptyStaticFunction);
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                aStaticDelegate();
+    }
+
+    [Benchmark]
+    public static void MeasureEvents()
+    {
+        Class aClass = new Class();
+        aClass.AnEvent += new MyDelegate(aClass.EmptyInstanceFunction);
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                aClass.MeasureFire100();
+    }
+
+    [Benchmark]
+    public static void GenericClassWithIntGenericInstanceField()
+    {
+        GenericClass<int> aGenericClassWithInt = new GenericClass<int>();
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                aGenericClassWithInt.aGenericInstanceFieldT = 1;
+    }
+
+    [Benchmark]
+    public static void GenericClassGenericStaticField()
+    {
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                GenericClass<int>.aGenericStaticFieldT = 1;
+    }
+
+    [Benchmark]
+    public static int GenericClassGenericInstanceMethod()
+    {
+        GenericClass<int> aGenericClassWithInt = new GenericClass<int>();
+
+        int x = 0;
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                x = aGenericClassWithInt.ClassGenericInstanceMethod();
+
+        return x;
+    }
+
+    [Benchmark]
+    public static int GenericClassGenericStaticMethod()
+    {
+        int x = 0;
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                x = GenericClass<int>.ClassGenericStaticMethod();
+
+        return x;
+    }
+
+    [Benchmark]
+    public static int GenericGenericMethod()
+    {
+        // Warmup
+        int x = 0;
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                x = Class.GenericMethod<int>();
+
+        return x;
+    }
+
+    [Benchmark]
+    public static void GenericClassWithSTringGenericInstanceMethod()
+    {
+        GenericClass<string> aGenericClassWithString = new GenericClass<string>();
+        string aString = "foo";
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                aGenericClassWithString.aGenericInstanceFieldT = aString;
+    }
+
+    [Benchmark]
+    public static int ForeachOverList100Elements()
+    {
+        List<int> iList = new List<int>();
+        for (int i = 0; i < 100; i++)
+            iList.Add(i);
+
+        int iResult = 0;
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                foreach (int i in iList)
+                    iResult = i;
+
+        return iResult;
+    }
```

Comment:
As I understand it, there is code generation and struct allocation that is specific to foreach regardless of the type.  ```List<T>``` just happens to be used in this case as these tests originated from desktop where the surface/runtime split did not exist - we tested everything.  It's true that this does happen to test some ```List<T>``` code as well, but that's just incidental.